endpoint.go

```diff
@@ -342,17 +442,29 @@ func (ep *endpoint) Leave(containerID string, options ...EndpointOption) error {
 }
 
 func (ep *endpoint) Delete() error {
-	var err error
-
 	ep.Lock()
-	epid := ep.id
-	name := ep.name
 	if ep.container != nil {
 		ep.Unlock()
-		return &ActiveContainerError{name: name, id: string(epid)}
+		return &ActiveContainerError{name: ep.name, id: string(ep.id)}
+	}
+	ep.Unlock()
+
+	if err := ep.deleteEndpoint(); err != nil {
+		return err
 	}
 
+	if err := ep.network.ctrlr.deleteEndpointFromStore(ep); err != nil {
```

Comment:
Should retrieve network and controller holding the respective locks and then `deleteEndpointFromStore` can be called